components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtGoogleAnalyticsTrackingHandler.java

```diff
@@ -243,57 +265,100 @@ private void trackPageView(MessageContext msgCtx) throws Exception {
+        String pageLocation = host + documentPath;
+
+        PageViewEvent pageViewEvent = new PageViewEvent();
+        pageViewEvent.setPageTitle(pageTitle);
+        pageViewEvent.setPageLocation(pageLocation);
+        pageViewEvent.setUserAgent(userAgent);
+        pageViewEvent.putParam(USER_IP, userIP);
+        pageViewEvent.putParam(GoogleAnalytics4Constants.ENGAGEMENT_TIME_MSEC_PARAM, "1");
+
+        GoogleAnalytics4Data data = new GoogleAnalytics4Data(apiSecret, measurementId);
+        data.setClientId(visitorId);
+        data.addEvent(pageViewEvent);
+
+        if (log.isDebugEnabled()) {
+            log.debug("Publishing https POST from gateway to Google analytics in GA4 format with ID: "
+                    + msgCtx.getMessageID() + " started at "
+                    + new SimpleDateFormat("[yyyy.MM.dd HH:mm:ss,SSS zzz]").format(new Date()));
+        }
+        boolean status = GoogleAnalytics4DataPublisher.publishData(data, userAgent);
+        if (log.isDebugEnabled()) {
+            log.debug("Publishing https POST from gateway to Google analytics in GA4 format with ID: "
+                    + msgCtx.getMessageID() + " ended " + (status ? "successfully at " : "with failure at ")
+                    + new SimpleDateFormat("[yyyy.MM.dd HH:mm:ss,SSS zzz]").format(new Date()));
+        }
+    }
+
+    private static String constructPageTitleForRequestPath(String httpMethod, String requestPath) {
+
+        return httpMethod + " " + requestPath;
+    }
+
+    /**
+     * A string is empty in our terms, if it is null, empty or a dash.
+     */
+    private static boolean isEmpty(String in) {
+
+        return in == null || "-".equals(in) || "".equals(in);
+    }
+
+    /**
+     * Generate a visitor id for this hit. If there is a visitor id in the
+     * messageContext, use that. Otherwise use a random number.
+     */
+    private static String getVisitorId(String account, String userAgent, MessageContext msgCtx)
+            throws NoSuchAlgorithmException, UnsupportedEncodingException {
+
+        if (msgCtx.getProperty(COOKIE_NAME) != null) {
+            return (String) msgCtx.getProperty(COOKIE_NAME);
+        }
+        String message;
+
+        AuthenticationContext authContext = APISecurityUtils.getAuthenticationContext(msgCtx);
+        if (authContext != null) {
+            message = authContext.getApiKey();
+        } else {
+            message = ANONYMOUS_USER_ID;
+        }
+
+        MessageDigest m = MessageDigest.getInstance("MD5");
+        m.update(message.getBytes("UTF-8"), 0, message.length());
+        byte[] sum = m.digest();
+        BigInteger messageAsNumber = new BigInteger(1, sum);
+        String md5String = messageAsNumber.toString(16);
+
+        /* Pad to make sure id is 32 characters long. */
+        while (md5String.length() < 32) {
+            md5String = "0" + md5String;
+        }
+
+        return "0x" + md5String.substring(0, 16);
+    }
```

Comment:
Looks like the old indentation has been removed and this has been added instead.
What is the correct indentation in this class, the new one or the old one?
And is everything aligned in the same way?